### PR TITLE
this includes upgrades up to 2.1.x elgg version + other fixes

### PR DESCRIPTION
**Pull request by propertunist**
mainly this brings videolist up to 2.1.x level compatibility and adds support for the google/youtube API.

### CODE REVIEW

File: classes/Videolist/Platform/Youtube.php [hunk 1]
```diff
@@ -24,22 +24,28 @@ public function parseUrl($url)
 		return false;
     }
 
-    public function getData($parsed)
+  public function getData($parsed)
     {
+        $API_key = elgg_get_plugin_setting('google_API_key', 'videolist');
         $video_id = $parsed['video_id'];
 
-		$xml = videolist_fetch_xml('http://gdata.youtube.com/feeds/api/videos/'.$video_id);
```
Comment (mrclay) [inline classes/Videolist/Platform/Youtube.php]: Does this just not work anymore?

Comment (mrclay) [inline classes/Videolist/Platform/Youtube.php]: I'm just thinking of people who upgrade, they'd need to be reminded to get an API key. Doesn't have to block this though.

Comment (propertunist) [inline classes/Videolist/Platform/Youtube.php]: yes, the old method is deactivated by google, an api key is required.

**Comment by juho-jaakkola** [comment]
Ok, so how to resolve the conflics.

Go through the files you've modified, and look for parts that have syntax like this:
```
<<<<<HEAD
    $foo = bar();
=====
    $foo = something_else_than_bar();
>>>>>your_branch
    $baz = baz();
```

This means that both you and the current master have modified the same file, and git does not know, which of the changes is the one it should be using.

Therefore you must manually choose which part of the code to use (the original or your version), and then remove the unnecessary part and also the ```<<<<<``, `======` and `>>>>>` lines from it.

So the previous example could look something like this after you have made the manual changes:
```
<?php
    $foo = something_else_than_bar();
    $baz = baz();
```

After this you must commit the fixes and add them to the pull request:
```shell
# Add the files, you've fixed:
git add path/to/file.php path/to/another/file.php

# Commit them:
git commit

# Then push the updated branch to github:
git push -f {your_remote} {your_branch}
```
The PR should now get updated automatically.

Further reading: http://weblog.masukomi.org/2008/07/12/handling-and-avoiding-conflicts-in-git/

**Comment by propertunist** [comment]
my local files don't have any such conflict symbols - which is why i didn't already resolve the issue. do i need to download from github and then fix locally?

**Comment by juho-jaakkola** [comment]
I have now created separate branches for `1.8`, `1.9` and `master` (Elgg 2), and I've added each of your commit/update to the lowest supported branch.

I wasn't able to add all of the changes, though. Can you review the changes that are missing, and make separate pull requests for the missing ones?

In the future, please create a new branch for each individual feature/bugfix. Also, never make changes directly to your own `master` branch if the repo is a fork from someone else's plugin.

Your local master branch has now been diverged from the official master, so you must fix your local repo:
```shell
# Make sure you're on the master branch
git checkout master
# Create a backup of your local master (because it still
# has some changes not added to the official repo)
git checkout -b my_changes
# Add the official repo as remote (if not there yet)
git remote add upstream https://github.com/Elgg/videolist.git
# Fetch latest changes (the ones I made) from the official repo
git fetch upsteam
# Reset your local master to match the official master
git checkout master
git reset --hard upstream/master
```

File: actions/videolist/get_metadata_from_url.php [hunk 2]
```diff
@@ -19,13 +20,24 @@
...
+                {
+                    //error_log('platform data= ' . print_r($platform_data,true));
+                    $result = array(
+                            'error' => false,
+                            'data' => array(
+                                    'title' => $platform_data['title'],
+                                    'description' => $platform_data['description'],
+                            ),
+                    );
+                }
```
Comment (juho-jaakkola) [inline actions/videolist/get_metadata_from_url.php]: This code block hasn't yet been added.

File: classes/Videolist/Platform/Youtube.php [hunk 3]
```diff
@@ -24,22 +24,28 @@ public function parseUrl($url)
...
+        }
+        else
+            {
+                register_error(elgg_echo('videolist:error:empty_provider_data'));
+                error_log('youtube video does not exist: ' . $video_id);
+                return false;
+            }
     }
-}
+}
```
Comment (juho-jaakkola) [inline classes/Videolist/Platform/Youtube.php]: Changes in this file haven't yet been added.

File: views/default/plugins/videolist/settings.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    echo "</h3>";    
+    echo '<br/><br/>';
+    echo "<h4>";
+    echo elgg_echo('videolist:admin:youtube_api') . ': ';
+    echo "</h4>";
+    echo elgg_view('input/text', array(
+                'name' => 'params[google_API_key]',
+                'value' => $google_api_key));
+    echo '<br/><br/>';
+
```
Comment (juho-jaakkola) [inline views/default/plugins/videolist/settings.php]: This file hasn't yet been added.

File: views/default/videolist/watch/vimeo.php [hunk 5]
```diff
@@ -1,4 +1,4 @@
 <?php
 
-$vars['src'] = "http://player.vimeo.com/video/" . $vars['entity']->video_id . "?byline=0";
+$vars['src'] = "//player.vimeo.com/video/" . $vars['entity']->video_id . "?byline=0";
```
Comment (juho-jaakkola) [inline views/default/videolist/watch/vimeo.php]: This is still missing.

File: views/default/videolist/watch/youtube.php [hunk 6]
```diff
@@ -1,4 +1,4 @@
 <?php
 
-$vars['src'] = "https://www.youtube-nocookie.com/embed/" . $vars['entity']->video_id;
+$vars['src'] = "https://www.youtube-nocookie.com/embed/" . $vars['entity']->video_id . "?html5=1";
```
Comment (juho-jaakkola) [inline views/default/videolist/watch/youtube.php]: This is still missing.

**Pull request closed without merging**